variables.tf

```diff
@@ -105,6 +105,11 @@ variable "local_cache_dir" {
   description = "Path on nodes for caching"
 }
 
+variable "build_job_hostmounts" {
+  description = "A map of name:path for which each named path will cause the host path to be mounted at the identical mountpoint in the build container."
+  default     = {}
```

Comment:
THis should be a list of maps

Reply:
@chickenandpork 
this should be 
```
default = []
type = list(map(any))
```

Reply:
My intent was to allow the key become the name of the map; this gives a somewhat trivial self-documenting of which rule creates a given hostmount (in case it's not obvious) and -- for me -- allows tying back a config piece to a related feature or function (in my example, "dogstatsd" is the key, and the host mount is used to host mount to get access to the dogstatsd Unix Domain Socket)

If optional types were permitted, I'd use this:
```
variable "build_job_hostmounts" {
  ...
  default     = {}
  type = map(object({
     host_path      = string
     container_path = optional(string)
     read_only      = optional(bool)
  }))
}
```

This allows me to write:
```
  build_job_hostmounts = {
    dogstatsd = {
      host_path = "/var/run/datadog"
    },
    some_other_mount = {
      host_path = "/some/path"
      container_path = "/other/path"
      read_only = true
    },
  }
```
The `dogstatsd` appears in the config.toml as:
```
                          [[runners.kubernetes.volumes.host_path]]
                            name = "dogstatsd"
                            mount_path = "/var/run/datadog"
                            host_path = "/var/run/datadog"
                            read_only = false
```

Reply:
that makes more sense. But optional vars I think were added 0.14; so we cannot use that. In that case, add an example in the README of using this and also we'll have to make the description as explanatory as possible. 

The description says list, but ideally should be a map of maps. We could set the type as: 
```hcl
type = map(map(string))
```
although, I'm not sure if that'll work.